apps/barista-design-system/src/pages/single-page/components/toc/toc.html

```diff
@@ -4,42 +4,48 @@
     (click)="_expandTocMenu()"
     [ngClass]="{ 'ba-is-active': _expandToc }"
   >
-    <a
-      href="#"
+    <p
       class="ba-toc-header ba-expandable-trigger ba-expandable-trigger-right"
       data-initialized="true"
-      >On this page</a
     >
+      On this page
+    </p>
     <div class="ba-toc-content ba-expandable-content">
       <ul class="ba-toc-list ba-list">
         <li
-          *ngFor="let heading of _headings$ | async"
+          *ngFor="let headline of tocItems"
           class="ba-toc-item"
           [class.ba-toc-item-active]="
-            _activeItems && _activeItems[0] === heading
+            _activeItems &&
+            _activeItems[0] &&
+            _activeItems[0].id === headline.id
           "
         >
-          <a
+          <span
             #headline
-            href="{{ heading.href }}"
+            [id]="headline.id"
```

Comment:
you are duplicating the id's on the page here. If there is a headline with this id, I don't think you will need to bind the id on this one here.
Sorry, this was my bad on the last review.